DigitalLearningSolutions.Data/DataServices/UserDataService/UserCentreDetailsDataService.cs

```diff
@@ -159,6 +159,24 @@ FROM UserCentreDetails
             ).SingleOrDefault();
         }
 
+        public IEnumerable<(string centreName, string? centreSpecificEmail)> GetAllCentreEmailsForUser(int userId)
+        {
+            return connection.Query<(string, string?)>(
+                @"SELECT c.CentreName, ucd.Email
+                    FROM DelegateAccounts AS da
+                    INNER JOIN Centres AS c ON c.CentreID = da.CentreID
+                    LEFT JOIN UserCentreDetails AS ucd ON ucd.UserID = da.UserID AND ucd.CentreID = c.CentreID
+                    WHERE da.UserID = @userId
+                    UNION
+                    SELECT c.CentreName, ucd.Email
+                    FROM AdminAccounts AS aa
+                    INNER JOIN Centres AS c ON c.centreID = aa.CentreID
+                    LEFT JOIN UserCentreDetails AS ucd ON ucd.UserID = aa.UserID AND ucd.CentreID = c.CentreID
+                    WHERE aa.UserID = @userId",

```

Comment:
I don't think so, because this won't take the cases where `ucd` doesn't contain anything for the given `userID` and `centreID`. For those cases, I want to select the centre name and email = null and then display the primary email on the page instead

Reply:
Ah yeah